docs/mlir.md

```diff
@@ -0,0 +1,25 @@
+# MLIR in MQT
+
+This part of MQT explores the capabilities of the Multi-Level Intermediate Representation (MLIR) in the context of compilation for quantum computing.
+We define multiple dialects, each with its dedicated purpose.
+For example, the MQTopt dialect is designed for optimization of quantum programs and features value-semantics.
```

Comment:
```suggestion
For example, the MQTOpt dialect is designed for optimization of quantum programs and features value-semantics.
```
Nitpick, but I think we mostly refer to it as `MQTOpt`.
Maybe something to add to the spelling pre-commit check that prohibits incorrect capitalisation? Same for `MQTDyn` and other potential candidates.